determinism.md

```diff
@@ -0,0 +1,174 @@
+GetDeterministicKey
```

Comment:
Code is compiler is being handled by @jaredpar .  This PR demonstrates how we intend to use it.  This PR is blocked until compiler adds this API.